printing/proof_diffs.ml

```diff
@@ -46,36 +46,37 @@ let write_color_enabled enabled =
 
 let color_enabled () = !term_color
 
-let diff_option = ref `OFF
+type diffOpt = DiffOff | DiffOn | DiffRemoved
 
-let read_diffs_option () = match !diff_option with
-| `OFF -> "off"
-| `ON -> "on"
-| `REMOVED -> "removed"
+let diffs_to_string = function
+  | DiffOff -> "off"
+  | DiffOn -> "on"
+  | DiffRemoved -> "removed"
 
-let write_diffs_option opt =
-  let enable opt =
-    if not (color_enabled ()) then
-      CErrors.user_err Pp.(str "Enabling Diffs requires setting the \"-color\" command line argument to \"on\" or \"auto\".")
-    else
-      diff_option := opt
-  in
-  match opt with
-  | "off" -> diff_option := `OFF
-  | "on" -> enable `ON
-  | "removed" -> enable `REMOVED
+
+let assert_color_enable () =
+  if not (color_enabled ()) then
+    CErrors.user_err
+      Pp.(str "Enabling Diffs requires setting the \"-color\" command line argument to \"on\" or \"auto\".")
+
+let string_to_diffs = function
+  | "off" -> DiffOff
+  | "on" -> assert_color_enable (); DiffOn
+  | "removed" -> assert_color_enable (); DiffRemoved
   | _ -> CErrors.user_err Pp.(str "Diffs option only accepts the following values: \"off\", \"on\", \"removed\".")
 
-let () =
-  Goptions.(declare_string_option {
-    optdepr = false;
-    optkey = ["Diffs"];
-    optread = read_diffs_option;
-    optwrite = write_diffs_option
-  })
+let opt_name = ["Diffs"]
```

Comment:
```suggestion
let proof_diffs_opt_name = ["Diffs"]
```

At the moment, it's not possible to grep for all the option keys in the source.  Perhaps there could be a wrapper function `opt_name` that could be used everywhere to make the option keys easy to grep.  If a new type is defined wrapping the key, the compiler could enforce using the wrapper.

Extra credit for defining separate wrapper functions for flags, options and tables (to make them easily-greppable).  Or maybe that's overkill.

Reply:
The reason why this is only `opt_name` and not `proof_diffs_opt_name` is because from outside this will be accessed as `Proof_diffs.opt_name`.

I didn't really grasp the rest of your comment.

Reply:
OK.

For the rest: If all keys appear in the code in the form `opt_key_wrapper(["Diffs"]` then grepping for "opt_key_wrapper" gets you a list of all flags, options and tables very simply.  Currently it's much more tedious to find them all.  Kapish?

Reply:
Yeah, I think I get it. OTOH, a cleaner solution would be for Coq to be able to dump information about all the defined options and flags, like what Emilio proposed in #11901 for warnings. This shouldn't be very difficult since `Print Options` already exists (this would be the same thing but with some machine-targeted output).